### PR TITLE
AN-5365/Streamline 2.0 Migration

### PR DESCRIPTION
**Pull request by forgxyz**
Updates project to Streamline 2.0

Legacy views and models namespaced to `v1` either in model name, directory, or both.
New set of streamline realtime and complete models for v2.

The request structure has been modified to simply send one request per block with verbosity = 2 in favor of two separate requests. This allows the addition of an `is_pending` check before processing any transactional data, which was leading to errors when a block fork occurred.

For deployment, ensure:
 - The API Integration and new UDFs are properly deployed
 - The Snowflake User / Role has the proper grants on prod
 - The latest FSC-Utils and Livequery-Models have been deployed
   - `dbt run -s livequery_models.deploy.core --vars '{UPDATE_UDFS_AND_SPS: true}'`
 - The new `v2` external tables are deployed, (https://github.com/FlipsideCrypto/streamline-snowflake/pull/186)
 - Rebuild the bronze views to point to these new buckets, `tag:streamline_helper` will do that
 - Full-refresh the streamline complete models, `tag:streamline_complete`

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ models:
     relation: true
     columns: true
   +on_schema_change: "append_new_columns"
+  bitcoin_models:
+    +pre-hook: '{{ fsc_utils.set_query_tag() }}'
 
 vars:
```
Comment (eric-laurello) [inline dbt_project.yml:44]: clean up double vars

File: dbt_project.yml [hunk 2]
```diff
@@ -70,3 +78,43 @@ dispatch:
...
+  API_INTEGRATION: '{{ var("config")[target.name]["API_INTEGRATION"] }}' 
+  EXTERNAL_FUNCTION_URI: '{{ var("config")[target.name]["EXTERNAL_FUNCTION_URI"] }}'
+  API_AWS_ROLE_ARN: '{{ var("config")[target.name]["API_AWS_ROLE_ARN"] if var("config")[target.name] else var("config")["dev"]["API_AWS_ROLE_ARN"] }}'
+  ROLES: '{{ var("config")[target.name]["ROLES"] }}'
+
+
+  config:
+  # The keys correspond to dbt profiles and are case sensitive
+    dev:
+      API_INTEGRATION: AWS_BITCOIN_API_STG
```
Comment (eric-laurello) [inline dbt_project.yml:104]: did shah deploy these integrations? I thought all the new ones were named with a _V2 on them

File: macros/streamline/streamline_udfs.sql [hunk 3]
```diff
@@ -21,3 +21,15 @@
...
     {%- endif %};
 {% endmacro %}
+
+{% macro create_udf_bulk_rest_api_v2() %}
+    CREATE
+    OR REPLACE EXTERNAL FUNCTION streamline.udf_bulk_rest_api_v2(
+        json OBJECT
+    ) returns ARRAY api_integration = 
+    {% if target.name == "prod" %}
+        aws_bitcoin_api AS 'https://6fmgkfdhy6.execute-api.us-east-1.amazonaws.com/prod/udf_bulk_rest_api'
```
Comment (eric-laurello) [inline macros/streamline/streamline_udfs.sql:31]: double check these integration names

File: models/streamline/realtime/streamline__blocks_hash_realtime.sql [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            'params',
+            ARRAY_CONSTRUCT(
+                block_number
+            )
+        ),
+        'Vault/prod/bitcoin/quicknode'
+    ) AS request
+FROM
+    blocks
+ORDER BY
```
Comment (eric-laurello) [inline models/streamline/realtime/streamline__blocks_hash_realtime.sql:94]: we can remove this and use this new sl param. Something like 
` "order_by_column": "-block_number" `

File: models/streamline/streamline__blocks.sql [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        'crosschain_silver',
+        'number_sequence'
+    ) }}
+WHERE
+    _id <= (
+        SELECT
+            MAX(block_number)
+        FROM
+            {{ ref('streamline__chainhead') }}
+    )
```
Comment (eric-laurello) [inline models/streamline/streamline__blocks.sql:19]: I would add a union all with a select 0 to make sure we pull that. the number table doesn't have a 0

File: dbt_project.yml (hunk 2 above)
Comment (forgxyz) [reply dbt_project.yml:104]: Can ask, changes brought in from https://github.com/FlipsideCrypto/bitcoin-models/pull/104/files

File: macros/streamline/api_integrations.sql [hunk 6]
```diff
@@ -1,15 +1,15 @@
 {% macro create_aws_bitcoin_api() %}
     {% if target.name == "prod" %}
```
[the remark this answers is not in the document]
Comment (forgxyz) [reply macros/streamline/api_integrations.sql:2]: @eric-laurello i.e. delete `api_integrations.sql`?

[the remark this answers is not in the document]
Comment (eric-laurello) [reply macros/streamline/api_integrations.sql:2]: yeah

File: dbt_project.yml (hunk 2 above)
Comment (forgxyz) [reply dbt_project.yml:91]: Yes, they should. Will update and deploy

**Pull request merged**